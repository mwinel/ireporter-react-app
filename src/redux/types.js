export const signupTypes = {
  SIGNUP_SUCCESS: "SIGNUP_SUCCESS"
};

export const loginTypes = {
  LOGIN_SUCCESS: "LOGIN_SUCCESS"
};
