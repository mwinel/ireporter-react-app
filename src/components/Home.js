import React, { Component } from "react";
import Navbar from '../components/Layout/Navbar';
import Footer from '../components/Layout/Footer';

class Home extends Component {
  render () {
    return (
      <div>
        <Navbar />
        <div className="container">
          <div className="row">
            <div className="flag-content">
              <h1>Welcome to iReporter!</h1>
            </div>
          </div>
          <Footer />
        </div>
      </div>
    );
  }
}

export default Home;
