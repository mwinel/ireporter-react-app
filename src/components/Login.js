import React, { Component } from 'react';
import { connect } from 'react-redux';
import Navbar from './Layout/Navbar';
import Footer from './Layout/Footer';
import login from '../redux/actions/loginAction';

export class Login extends Component {
  constructor (props) {
    super(props);
    this.state = {
      username: '',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillMount () {
    this.props.login();
  }

  onChange (e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit (e) {
    e.preventDefault();
    const loginUser = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.login(loginUser);
  }

  render () {
    return (
      <div>
        <Navbar />
        <div className="row">
          <form id="userLogin" onSubmit={this.onSubmit}>
            <h2>Login to iReporter</h2>
            <input type="text" name="username" onChange={this.onChange} value={this.state.username} placeholder="Enter username" />
            <input type="text" name="password" onChange={this.onChange} value={this.state.password} placeholder="Enter password" />
            <button type="submit" className="btn-login">Login</button>
            <p className="form-footer">Don't have an account <a href="./signup">Sign Up</a>.</p>
          </form>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  login: state.login.login
});

export default connect(mapStateToProps, { login })(Login);
