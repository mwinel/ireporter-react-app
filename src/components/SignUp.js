import React, { Component } from 'react';
import { connect } from 'react-redux';
import Navbar from '../components/Layout/Navbar';
import Footer from '../components/Layout/Footer';
import signUp from '../redux/actions/signUpAction';

export class SignUp extends Component {
  constructor (props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      othernames: '',
      username: '',
      email: '',
      password: '',
      phone_number: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillMount () {
    this.props.signUp();
  }

  onChange (e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit (e) {
    e.preventDefault();
    const signup = {
      firstname: this.state.firstname,
      lastname: this.state.lastname,
      othernames: this.state.othernames,
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      phone_number: this.state.phone_number
    };
    this.props.signUp(signup);
  }

  render () {
    return (
      <div>
        <Navbar />
        <div className="row">
          <form id="addUser" onSubmit={this.onSubmit}>
            <h2>Please Sign Up</h2>
            <input type="text" name="firstname" onChange={this.onChange} value={this.state.firstname} placeholder="Enter firstname" />
            <input type="text" name="lastname" onChange={this.onChange} value={this.state.lastname} placeholder="Enter lastname" />
            <input type="text" name="othernames" onChange={this.onChange} value={this.state.othernames} placeholder="Enter othernames" />
            <input type="text" name="username" onChange={this.onChange} value={this.state.username} placeholder="Enter username" />
            <input type="text" name="email" onChange={this.onChange} value={this.state.email} placeholder="Enter email" />
            <input type="text" name="password" onChange={this.onChange} value={this.state.password} placeholder="Enter password" />
            <input type="text" name="phone_number" onChange={this.onChange} value={this.state.phone_number} placeholder="Enter phone number" />
            <button type="submit" className="btn-signup">Sign Up</button>
            <p className="form-footer">Already have an account <a href="./login">Login</a>.</p>
          </form>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user.user
});

export default connect(mapStateToProps, { signUp })(SignUp);
