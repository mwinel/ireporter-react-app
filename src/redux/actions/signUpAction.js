import { signupTypes } from '../types';
import { signupUrl } from '../../urls';

const signUp = userData => dispatch => {
  return fetch(signupUrl, {
    method: 'POST',
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(userData)
  })
    .then(response => response.json())
    .then(data => {
      dispatch({
        type: signupTypes.SIGNUP_SUCCESS,
        payload: data
      });
      if (data.message) {
        alert(`${data.message}`);
        window.setTimeout(function () {
          window.location.replace('./login');
          window.location.reload(true);
        }, 1000);
      } else if (data.error) {
        alert(`${data.error}`);
      }
    });
};

export default signUp;
