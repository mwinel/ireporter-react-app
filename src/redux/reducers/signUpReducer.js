import { signupTypes } from '../types';

const initialState = {
  user: {},
  errors: {}
};

export default function (state = initialState, action) {
  switch (action.type) {
  case signupTypes.SIGNUP_SUCCESS:
    return {
      ...state,
      user: action.payload
    };
  default:
    return state;
  }
}
