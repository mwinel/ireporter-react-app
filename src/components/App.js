import React, { Component } from "react";
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import '../styles/App.css';
import store from '../redux/store';
import Home from '../components/Home';
import SignUp from '../components/SignUp';
import Login from '../components/Login';

class App extends Component {
  render () {
    return (
      <BrowserRouter>
        <Provider store={store}>
          <Route exact path="/" component={Home} />
          <Route exact path="/signup" component={SignUp} />
          <Route exact path="/login" component={Login} />
        </Provider>
      </BrowserRouter>
    );
  }
}

export default App;
