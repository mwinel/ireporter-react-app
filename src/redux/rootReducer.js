import { combineReducers } from 'redux';
import indexReducer from './reducers/indexReducer';
import signUpReducer from './reducers/signUpReducer';
import loginReducer from './reducers/loginReducer';

export default combineReducers({
  test: indexReducer,
  user: signUpReducer,
  login: loginReducer
});
