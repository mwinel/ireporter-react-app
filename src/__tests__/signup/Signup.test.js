import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import fetchMock from 'fetch-mock'
import signUpReducer from '../../redux/reducers/signUpReducer';
import signUpAction from '../../redux/actions/signUpAction';
import { SignUp } from '../../components/SignUp';
import { signupTypes } from '../../redux/types';

const storeItems = {};
const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const store = mockStore(storeItems);

const initialState = {};

const props = {
  firstname: '',
  lastname: '',
  othernames: '',
  username: '',
  email: '',
  password: '',
  phone_number: '',
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  signUp: jest.fn(),
  preventDefault: jest.fn()
}

const mockEvent = {
  preventDefault: jest.fn()
}

describe('SignUp Component', () => {
  let SignUpComponent = shallow(<SignUp {...props} store={store}/>)
  it('Should match snapshot', () => {
    SignUpComponent.instance().onSubmit(mockEvent);
    expect(toJson(SignUpComponent)).toMatchSnapshot();
  });

  it('should trigger onChange on signup', () => {
    const signup = { 
      firstname: 'hahahahahha',
      lastname: 'ahahahahahhaha',
      othernames: 'aahahahahhahaha',
      username: 'mwinelnnnnn',
      email: 'kisuulejames1@gmail.com',
      password: 'password',
      phone_number: '123456',
    }
    const evt = {
      preventDefault: jest.fn(),
      target: signup
    }
    expect(SignUpComponent.instance().onChange(evt));
  });
});

describe("signup success", () => {
  it('tests signup success', () => {
    const newState = signUpReducer(initialState,
      { type: signupTypes.SIGNUP_SUCCESS });
    expect(newState).toEqual({});
  });
});

describe("signup action", () => {
  afterEach(() => {
    fetchMock.restore()
  })

  it("creates SIGNUP_SUCCESS when creating user", () => {
    fetchMock.post('*', {
      body: {
        firstname: 'hahahahahha',
        lastname: 'ahahahahahhaha',
        othernames: 'aahahahahhahaha',
        username: 'mwinelnnnnn',
        email: 'kisuulejames1@gmail.com',
        password: 'password',
        phone_number: '123456',
      },
      headers: { 'content-type': 'application/json' }
    })

    const expectedActions = [
      { type: signupTypes.SIGNUP_SUCCESS, 
        payload: {
          firstname: 'hahahahahha',
          lastname: 'ahahahahahhaha',
          othernames: 'aahahahahhahaha',
          username: 'mwinelnnnnn',
          email: 'kisuulejames1@gmail.com',
          password: 'password',
          phone_number: '123456',
        }
      }
    ]
    return store.dispatch(signUpAction()).then(() => {
      expect(store.getActions()).toEqual(expectedActions)
    })
  })
})
