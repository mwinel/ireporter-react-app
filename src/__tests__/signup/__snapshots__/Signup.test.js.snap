// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component Should match snapshot 1`] = `
<div>
  <Navbar />
  <div
    className="row"
  >
    <form
      id="addUser"
      onSubmit={[Function]}
    >
      <h2>
        Please Sign Up
      </h2>
      <input
        name="firstname"
        onChange={[Function]}
        placeholder="Enter firstname"
        type="text"
        value=""
      />
      <input
        name="lastname"
        onChange={[Function]}
        placeholder="Enter lastname"
        type="text"
        value=""
      />
      <input
        name="othernames"
        onChange={[Function]}
        placeholder="Enter othernames"
        type="text"
        value=""
      />
      <input
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        type="text"
        value=""
      />
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter email"
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="text"
        value=""
      />
      <input
        name="phone_number"
        onChange={[Function]}
        placeholder="Enter phone number"
        type="text"
        value=""
      />
      <button
        className="btn-signup"
        type="submit"
      >
        Sign Up
      </button>
      <p
        className="form-footer"
      >
        Already have an account 
        <a
          href="./login"
        >
          Login
        </a>
        .
      </p>
    </form>
  </div>
  <Footer />
</div>
`;
